src/sources/socket/udp.rs

```diff
@@ -75,12 +75,11 @@ pub struct UdpConfig {
     receive_buffer_bytes: Option<usize>,
 
     #[configurable(derived)]
-    #[serde(default = "default_framing_message_based")]
-    pub(super) framing: FramingConfig,
+    pub(super) framing: Option<FramingConfig>,
```

Comment:
Changed the `UdpConfig.framing` field to allow Optional framing, so we can compute the framing default based on the deserialized  ( using the function `default_framing_message_based from the `DeserializerConfig`)